parachain/pallets/ethereum2-light-client/src/lib.rs

```diff
@@ -0,0 +1,165 @@
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+        /// https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/sync-protocol.md#constants
+        /// Finalized root index - TODO not a useful comment, will elaborate as understanding grows
+		#[pallet::constant]
+		type FinalizedRootIndex: Get<u16>;
+		/// Next sync committee index - TODO not a useful comment, will elaborate as understanding grows
+		#[pallet::constant]
+		type NextSyncCommitteeIndex: Get<u16>;
+        /// Weight information for extrinsics in this pallet
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::event]
+	pub enum Event<T> {}
+
+	#[pallet::error]
+	pub enum Error<T> {
+        // TODO: Add
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+    // https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/sync-protocol.md#lightclientstore
+    /// Beacon block header that is finalized
+    #[pallet::storage]
+	pub(super) type FinalizedHeader<T: Config> = StorageValue<_, BeaconBlockHeader, ValueQuery>;
+
+    /// Current sync committee corresponding to the active header
+    #[pallet::storage]
+    pub(super) type CurrentSyncCommittee<T: Config> = StorageValue<_, SyncCommittee, ValueQuery>;
+
+    /// Next sync committee corresponding to the active header
+    #[pallet::storage]
+    pub(super) type NextSyncCommittee<T: Config> = StorageValue<_, SyncCommittee, ValueQuery>;
+
+    /// Best available header to switch finalized head to if we see nothing else
+    #[pallet::storage]
+    pub(super) type BestValidUpdate<T: Config> = StorageValue<_, BeaconBlockHeader, ValueQuery>;
+
+    /// Most recent available reasonably-safe header
+    #[pallet::storage]
+    pub(super) type OptimisticHeader<T: Config> = StorageValue<_, BeaconBlockHeader, ValueQuery>;
+
+    /// Max number of active participants in a sync committee (used to calculate safety threshold)
+    #[pallet::storage]
+    pub(super) type PreviousMaxActiveParticipants<T: Config> = StorageValue<_, u64, ValueQuery>;
+
+    #[pallet::storage]
+    pub(super) type CurrentMaxActiveParticipants<T: Config> = StorageValue<_, u64, ValueQuery>;
+
+    // Would these also go into the store?
+    // https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/sync-protocol.md#lightclientupdate
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig {
+		// genesis header goes header, maybe?
+	}
+
+	#[cfg(feature = "std")]
+	impl Default for GenesisConfig {
+		fn default() -> Self {
+			Self {}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(T::WeightInfo::import_header())]
```

Comment:
For now, I'd just ignore benchmarks & weights, and use a dummy constant:

```suggestion
		#[pallet::weight(1_000_000)]
```